core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1599,11 +1621,18 @@ private[spark] class DAGScheduler(
                   case mapStage: ShuffleMapStage =>
                     val numMissingPartitions = mapStage.findMissingPartitions().length
                     if (numMissingPartitions < mapStage.numTasks) {
-                      // TODO: support to rollback shuffle files.
-                      // Currently the shuffle writing is "first write wins", so we can't re-run a
-                      // shuffle map stage and overwrite existing shuffle files. We have to finish
-                      // SPARK-8029 first.
-                      abortStage(mapStage, generateErrorMessage(mapStage), None)
+                      if (sc.getConf.getBoolean("spark.shuffle.useOldFetchProtocol", false)) {
```

Comment:
can we define this config in `org.apache.spark.internal.config`? so that we don't need to hardcode the config name here.

Reply:
Copy that, also change the hard code in tests. Done in 5a4b8b8.